bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+# Extension BOLT XX: Taproot Gossip
+
+# Table of Contents
+
+  * [Aim](#aim)
+  * [Overview](#overview)
+  * [Terminology](#terminology)
+  * [Type Definitions](#type-definitions)
+  * [TLV Based Messages](#tlv-based-messages)
+  * [Taproot Channel Proof and Verification](#taproot-channel-proof-and-verification)
+  * [Timestamp fields](#timestamp-fields)
+  * [Bootstrapping Taproot Gossip](#bootstrapping-taproot-gossip)
+  * [Specification](#specification)
+    * [Feature Bits](#feature-bits)
+    * [The `announcement_signatures_2` message](#the-announcement_signatures_2-message)
+    * [The `channel_announcement_2` message](#the-channel_announcement_2-message)
+    * [The `node_announcement_2` message](#the-node_announcement_2-message)
+    * [The `channel_update_2` message](#the-channel_update_2-message)
+  * [Appendix A: Algorithms](#appendix-a-algorithms)
+      * [Partial Signature Calculation](#partial-signature-calculation)
+      * [Partial Signature Verification](#partial-signature-verification)
+  * [Appendix B: Test Vectors](#appendix-b-test-vectors)
+  * [Acknowledgements](#acknowledgements)
+
+## Aim 
+
+This document aims to update the gossip protocol defined in [BOLT 7][bolt-7] to
+allow for advertisement and verification of taproot channels. An entirely new
+set of gossip messages are defined that use [BIP-340][bip-340] signatures and
+that use a purely TLV based schema.
```

Comment:
You can't though, because that would not be backwards compatible :(  So you actually do have compulsory fields, which cannot be removed, until everyone upgrades.

It's actually simpler to define a new type in that case!